pytorch_lightning/utilities/fetching.py

```diff
@@ -257,10 +259,14 @@ def _fetch_next_batch(self, iterator: Iterator) -> None:
         self.on_fetch_end(batch, start_output)
 
     def move_to_device(self, batch: Any) -> Any:
-        if self.store_on_device and self.batch_to_device is not None:
+        if self.store_on_device:
             batch = self.batch_to_device(batch)
         return batch
 
+    def reset(self) -> None:
+        super().reset()
+        self.batches: List[Any] = []
```

Comment:
Can we add `batches: List[Any]` to the stub (like static attributes) if defined outside `__init__`? Then no need for typing here anymore :)